### ISSUE TITLE
Disable the "Reset styles" button when there are no changes.

### ISSUE THREAD

**Issue by emdashftw** [WordPress/gutenberg#63495]
## What problem does this address?
I've just did a styles reset and noticed the "Reset styles" button is still enabled.
![image](https://github.com/user-attachments/assets/f5ae41ef-f4fa-465e-89b4-8dbce14c4d04)

It let me reset again and It still says there's 1 site change to be saved.
![image](https://github.com/user-attachments/assets/0cadf3f6-658d-4a71-966a-2e5a8d0d707f)

Makes zero sense to me. And I find this behavior very confusing for the end user.
It goes against the user to properly identify when the styles are truly reset.
It requires you to remember if you've reset them.

## What is your proposed solution?
If the Global Styles are reset. The "Reset styles" button should be disabled.
![image](https://github.com/user-attachments/assets/b72027af-3620-430a-af75-6a093f804951)

Note that I've used the DevTools to make the text look Greyish (disabled).

**Comment by t-hamano** [comment]
Thanks for the report.

If the global styles hasn't changed, or if you've reset the global styles once, the button should be disabled. However, for some reason the `useGlobalStylesReset` hook isn't working properly and `canReset` always seems to be `true`.

https://github.com/WordPress/gutenberg/blob/48b835044028807694ae8a420994430787b4d55d/packages/block-editor/src/components/global-styles/hooks.js#L87-L94

The reason for this might be that even if the global styles hasn't changed, the two configs are different, so `fastDeepEqual` is always `false`.

`config`:

```
{
	settings: {},
	styles: {},
	_links: {
		self: [
			{
				href: 'http://localhost:8888/wp-json/wp/v2/global-styles/5',
			},
		],
		'version-history': [
			// ...
		],
		'wp:action-publish': [
			// ...
		],
		'wp:action-edit-css': [
			// ...
		],
		curies: [],
	},
}
```

`EMPTY_CONFIG`:

```
{
	settings: {},
	styles: {},
}
```

**Comment by t-hamano** [comment]
After some quick investigation, it appears that the newly added `_links` are involved, as removing the `_links` from this object resolves the issue:

https://github.com/WordPress/gutenberg/blob/48b835044028807694ae8a420994430787b4d55d/packages/editor/src/components/global-styles-provider/index.js#L89-L95

**Comment by t-hamano** [comment]
A simple fix would be to rewrite the `useGlobalStylesReset` hook as follows to exclude everything except `settings`  and `styles`, but there may be a better way.

```diff
diff --git a/packages/block-editor/src/components/global-styles/hooks.js b/packages/block-editor/src/components/global-styles/hooks.js
index 44a3d5e23e..81e57ca5dc 100644
--- a/packages/block-editor/src/components/global-styles/hooks.js
+++ b/packages/block-editor/src/components/global-styles/hooks.js
@@ -85,7 +85,11 @@ const VALID_SETTINGS = [
 ];
 
 export const useGlobalStylesReset = () => {
-       const { user: config, setUserConfig } = useContext( GlobalStylesContext );
+       const { user: _config, setUserConfig } = useContext( GlobalStylesContext );
+       const config = {
+               settings: _config.settings,
+               styles: _config.styles,
+       };
        const canReset = !! config && ! fastDeepEqual( config, EMPTY_CONFIG );
        return [
                canReset,
```

**Comment by andrewserong** [comment]
@t-hamano ignoring `_links` as in your diff above sounds good to me. The `_links` is essentially metadata so shouldn't be used for the reset comparison.

**Comment by t-hamano** [comment]
Fixed by #63562
